plugins/runc/internal/container/run_adapters.go

```diff
@@ -18,21 +19,16 @@ func LoadSpecFromBundle(next types.Run) types.Run {
 	return func(ctx context.Context, opts types.Opts, resp *daemon.RunResp, req *daemon.RunReq) (chan int, error) {
 		details := req.GetDetails().GetRunc()
 		bundle := details.GetBundle()
+		workingDir := details.GetWorkingDir()
 
-		// If empty, assume cwd is the bundle. This is the behavior of runc binary as well.
-		if bundle != "" {
-			cwd, err := os.Getwd()
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to get current working directory: %v", err)
-			}
-			err = os.Chdir(bundle)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to set working directory: %v", err)
-			}
-			defer os.Chdir(cwd)
```

Comment:
lgtm, but defer would be applied to all functions so are we handling, are all paths absolute that we pass to runc for libcontainer stuff?

Reply:
@swarnimarun Wdym by defer would be applied to all functions? Chdir would change the current directory of the entire process or thread, and not just this go routine thats handling the gRPC request.